Realm/RLMResults.mm

```diff
@@ -376,6 +376,9 @@ - (id)objectAtIndexedSubscript:(NSUInteger)index {
 }
 
 - (id)aggregate:(NSString *)property method:(util::Optional<Mixed> (Results::*)(size_t))method methodName:(NSString *)methodName {
+    if (_results.get_mode() == Results::Mode::Empty) {
```

Comment:
`-indexOfObjectWithPredicate:` having the check to avoid calling `Results::get_query` seems reasonable enough.

`-objectsWithPredicate:` and `-sortedResultsUsingDescriptors:` aren't consistent with that logic though. It should be perfectly valid to call `Results::filter` or `Results::sort` on an empty `Results`, yet they first check for `Mode::Empty`. ISTM that we should fix `Results::sort` and `Results::filter` to work when empty (presumably by returning another empty `Results`), and then switch `-objectsWithPredicate:` and `-sortedResultsUsingDescriptors:` to checking for `_info` being null instead.